code/modules/swarmers/swarmer_act.dm

```diff
@@ -178,13 +193,28 @@
 
 /obj/structure/window/swarmer_act(mob/living/simple_animal/hostile/swarmer/actor)
 	var/is_on_shuttle = istype(get_area(src), /area/shuttle)
+	var/pressure_greatest = 0
+	var/pressure_smallest = INFINITY 	//Freaking terrified to use INFINITY, man
+	for(var/t in RANGE_TURFS(2, src))	//Begin processing the delta pressure across the wall.
+		var/turf/open/turf_adjacent = t
+		if(!istype(turf_adjacent))
+			continue
+		if(turf_adjacent.air.return_pressure() > pressure_greatest)
+			pressure_greatest = turf_adjacent.air.return_pressure()
+		if(turf_adjacent.air.return_pressure() < pressure_smallest)
+			pressure_smallest = turf_adjacent.air.return_pressure()
+	var/delta_p = pressure_greatest - pressure_smallest
 	for(var/turf/adj_turf in range(1, src))
 		var/area/adj_area = get_area(adj_turf)
-		if(isspaceturf(adj_turf) || (!is_on_shuttle && (istype(adj_area, /area/shuttle) || istype(adj_area, /area/space))) || (is_on_shuttle && !istype(adj_area, /area/shuttle)))
+		if (delta_p > 1000) //If the delta pressure is too high it is considered too dangerous to break this. A difference of 1000 kPa is considered dangerous
+			to_chat(actor, "<span class='warning'>Destroying this object has the potential to cause an explosive pressure release. Aborting.</span>")
+			actor.target = null
+			return TRUE
+		else if(isspaceturf(adj_turf) || (!is_on_shuttle && (istype(adj_area, /area/shuttle) || istype(adj_area, /area/space))) || (is_on_shuttle && !istype(adj_area, /area/shuttle)))
```

Comment:
add a check for adj_turf.planetary_atmos here and it'll help prevent icebox breaches being a nonissue to swarmer code.